### PR TITLE
Fix the issue that first background files are in double precession while mpas-bundle in single-precession.

### PR DESCRIPTION
**Pull request by mos3r3n**
### Description
Our mpas-bundle is commonly built in single precession. However, the first background files are in double precession (i.e. ensemble background at 00Z 15 April 2018). In that case, the analysis output will report error information "ERROR: MPAS IO Error: Inconsistent redefinition of dimension" (@RGNystrom reported). To solve this issue, the NCO is employed to convert double (float) variables to float (double) variables when the mpas-bundle is built in single (double) precision.

#### Tier 1:
 - [x] 3dvar_OIE120km_WarmStart
 - [x] 4denvar_OIE120km_WarmStart
 - [x] letkf_OIE120km_WarmStart

### CODE REVIEW

File: bin/LinkWarmStartBackgrounds.csh [hunk 1]
```diff
@@ -28,8 +29,23 @@ while ( $member <= $nMembers )
...
       ln -sfv ${InitialMemberFC}/${firstbackground__filePrefixOuter}.*.nc $CyclingFCDirs[$member]/.
-      # rm ${fcFile}
-      cp ${fcFile}${OrigFileSuffix} ${fcFile}
+      # Handling double/single precision files
+      foreach file ($CyclingFCDirs[$member]/*.nc)
+        set varinfo = `ncdump -h ${file} | grep surface_pressure`
+        if ( "$varinfo" =~ "*double*" ) then
+           if ( "$bundlePrecision" == "single" ) then
+              mv ${file} ${file}${OrigFileSuffix}
+              ncpdq -5 --pck_map=dbl_flt ${file}${OrigFileSuffix} ${file}
```
Comment (jim-p-w) [inline bin/LinkWarmStartBackgrounds.csh:38]: Where can I find `ncpdq`? It's not in my path

Comment (mos3r3n) [reply bin/LinkWarmStartBackgrounds.csh:38]: It's a function of the nco. You can see I add "module load nco" in suiteBase.py.

**Review by RGNystrom — APPROVED** [review]

**Comment by jim-p-w** [comment]
Instead of converting the files on the fly every time this step is run would it be more efficient to convert the files once, and then use the appropriate files based on `bundlePrecision`?

**Comment by liujake** [comment]
@jim-p-w has a good point. It is better to simply convert files offline once, then use them in the workflow, e.g., replace the old double-precision file location with the new single-precision file location.

**Comment by mos3r3n** [comment]
That's right. Since we now use single-precision mpas-bundle only, it would be better that we prepare these files in single-precession rather than converting them when running workflow. @jim-p-w @liujake

**Comment by liujake** [comment]
@mos3r3n You can still use this PR, but fixing this in a different way: provide new location of the first cycle ensemble background files.

**Comment by liujake** [comment]
I think this one can be closed. @mos3r3n . We should have addressed this by converting double precision first background to single precision outside the workflow.

**Pull request closed without merging**